dev_tool/src/jumpbox.ts

```diff
@@ -191,6 +194,115 @@ async function ensureAllowlistIP(
     return undefined
 }
 
+// Function to securely prompt for password
+async function promptPassword(prompt: string): Promise<string> {
+    const rl = createInterface({ input, output })
+
+    return new Promise((resolve) => {
+        const stdin = process.stdin
+        const listener = (char: Buffer) => {
+            const charStr = char.toString()
+            switch (charStr) {
+                case '\n':
+                case '\r':
+                case '\u0004':
+                    stdin.removeListener('data', listener)
+                    stdin.setRawMode(false)
+                    stdin.pause()
+                    output.write('\n')
+                    rl.close()
+                    resolve(password)
+                    break
+                case '\u0003': // Ctrl+C
+                    stdin.removeListener('data', listener)
+                    stdin.setRawMode(false)
+                    stdin.pause()
+                    rl.close()
+                    process.exit()
+                    break
+                case '\u007f': // Backspace
+                    if (password.length > 0) {
+                        password = password.slice(0, -1)
+                        output.write('\b \b')
+                    }
+                    break
+                default:
+                    password += charStr
+                    break
+            }
+        }
+
+        let password = ''
+        output.write(prompt)
+        // Set raw mode and disable echo
+        stdin.setRawMode(true)
+        process.stdin.setRawMode(true)
+        const originalIsRaw = stdin.isRaw
+        const originalIsTTY = stdin.isTTY
+        stdin.setRawMode(true)
+        stdin.resume()
+
+        // Clean up on exit
+        const cleanup = () => {
+            stdin.setRawMode(originalIsRaw)
+            if (originalIsTTY) {
+                stdin.isTTY = originalIsTTY
+            }
+        }
+
+        stdin.on('data', listener)
+        rl.once('close', cleanup)
+    })
+}
+
+// Attempt SSH connection, prompting for password if needed
+async function attemptSSHConnection(
+    host: string,
+    username: string,
+    privateKeyPath: string
+): Promise<NodeSSH> {
+    const ssh = new NodeSSH()
+
+    try {
+        // Read the private key file
+        const privateKey = readFileSync(privateKeyPath, 'utf8')
+
+        try {
+            await ssh.connect({
+                host,
+                username,
+                privateKey,
+            })
+        } catch (err) {
+            if (
+                err instanceof Error &&
+                (err.message.includes('password') ||
```

Comment:
For my error message the keyword `password` was not included. Mine used `passphrase` and `Encrypted` with capital `E`. Maybe we always prompt for a password so we don't need to trying to catch this?

![Screenshot 2024-12-13 at 2 59 59 PM](https://github.com/user-attachments/assets/a9719d77-e670-4bf7-9cf2-5c51fc4c72fe)

Reply:
ok thanks, good catch! I'll update this.